### PR TITLE
image: add meta/gadget.yaml infrastructure

### PR DESCRIPTION
**Pull request by mvo5**
Gadget.yaml spec described here: https://gist.github.com/niemeyer/9acc19553655407c576f2728a855d7d6

### CODE REVIEW

**Comment by ogra1** [comment]
"bootloader: grub" but everything else uboot ? smells a bit confusing :)

**Comment by chipaca** [comment]
:+1:

File: snap/gadget.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+type GadgetInfo struct {
+	Bootloader string
+	Volumes    map[string][]Volume
+}
+
+func ReadGadgetInfo(info *Info) (*GadgetInfo, error) {
+	const errorFormat = "cannot read gadget snap details: %s"
+
+	gadgetYamlFn := filepath.Join(info.MountDir(), "meta", "gadget.yaml")
+	kmeta, err := ioutil.ReadFile(gadgetYamlFn)
```
Comment (pedronis) [inline snap/gadget.go]: nitpick, gmeta?

File: snap/gadget.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	for k, vl := range gy.Volumes {
+		gi.Volumes[k] = make([]Volume, len(vl))
+		for i, v := range vl {
+			gi.Volumes[k][i] = Volume{
+				Name:    v.Name,
+				Label:   v.Label,
+				Type:    v.Type,
+				Data:    v.Data,
+				Offset:  v.Offset,
+				Content: v.Content,
```
Comment (pedronis) [inline snap/gadget.go]: are all these values optional? is an empty volume entry valid?

**Comment by swem** [comment]
Will we include model assertion in gadget.yaml ? Or somewhere else in gadget snap?

**Comment by pedronis** [comment]
@swem the plan (as discussed last) is for the model assertion to be put in /var/lib/snapd/seed/assertions/model.assertion or something like that,  it is needed to double check the gadget and we plan to have device specific gadget snaps but also general/reused ones so it cannot live in the gadget snap itself

**Pull request merged**